components/style/properties/gecko.mako.rs

```diff
@@ -937,6 +937,32 @@ fn static_assert() {
 
 </%self:impl_trait>
 
+<%def name="simple_background_array_property(name, field_name)">
+    pub fn copy_background_${name}_from(&mut self, other: &Self) {
+        unsafe {
+            Gecko_EnsureImageLayersLength(&mut self.gecko.mImage, other.gecko.mImage.mLayers.len());
+        }
+        for (layer, other) in self.gecko.mImage.mLayers.iter_mut()
+                                  .zip(other.gecko.mImage.mLayers.iter())
+                                  .take(other.gecko.mImage.${field_name}Count as usize) {
+            layer.${field_name} = other.${field_name};
+        }
+        self.gecko.mImage.${field_name}Count = other.gecko.mImage.${field_name}Count;
+    }
+
+    pub fn set_background_${name}(&mut self, v: longhands::background_${name}::computed_value::T) {
+        unsafe {
+          Gecko_EnsureImageLayersLength(&mut self.gecko.mImage, v.0.len());
+        }
+
+        self.gecko.mImage.${field_name}Count = v.0.len() as u32;
+        for (servo, geckolayer) in v.0.into_iter().zip(self.gecko.mImage.mLayers.iter_mut()) {
+            geckolayer.${field_name} = {
+                ${caller.body()}
+            };
+        }
+    }
```

Comment:
This is still missing some pieces wrt cycling the gecko arrays, but it needs to be done after the cascading of all properties. Will fix.